lib/chef/resource/file/verification.rb

```diff
@@ -0,0 +1,123 @@
+      # preventing file reosurce content deploys.  Their intended use
+      # is to verify the contents of a temporary file before it is
+      # deployed onto the system.
+      #
+      # Similar to not_if and only_if, file verifications can take a
+      # ruby block, which will be called, or a string, which will be
+      # executed as a Shell command.
+      #
+      # Additonally, Chef or third-party verifications can ship
+      # "registered verifications" that the user can use by specifying
+      # a :symbol as the command name.
+      #
+      # To create a registered verification, create a class that
+      # inherits from Chef::Resource::File::Verification and use the
+      # register class method to give it name.  Registered
+      # verifications are expected to supply a verify instance method
+      # that takes 2 arguments.
+      #
+      # Example:
+      # class Chef
+      #  class Resource
+      #    class File::Verification::Foo < Chef::Resource::File::Verification
+      #      provides :noop
+      #      def verify(path, opts)
+      #        #yolo
+      #        true
+      #      end
+      #    end
+      #  end
+      # end
+      #
+      #
+
+      class Verification
+        extend Chef::Mixin::DescendantsTracker
+
+        def self.provides(name)
+          @provides = name
+        end
+
+        def self.provides?(name)
+          @provides == name
+        end
+
+        def self.lookup(name)
+          c = descendants.find {|d| d.provides?(name) }
+          if c.nil?
+            raise Chef::Exceptions::VerificationNotFound.new "No file verification for #{name} found."
+          end
+          c
+        end
+
+        def initialize(parent_resource, command, opts, &block)
+          @command, @command_opts = command, opts
+          @block = block
+          @parent_resource = parent_resource
+        end
+
+        def verify(path, opts={})
+          Chef::Log.debug("Running verification[#{self}] on #{path}")
+          if @block
+            verify_block(path, opts)
+          elsif @command.is_a?(Symbol)
+            verify_registered_verification(path, opts)
+          elsif @command.is_a?(String)
+            verify_command(path, opts)
+          end
+        end
+
+        def verify_block(path, opts)
+          @block.call(path)
+        end
+
+        # We reuse Chef::GuardInterpreter in order to support
+        # the same set of options that the not_if/only_if blocks do
+        def verify_command(path, opts)
+          command = @command % {:file => path}
+          interpreter = if @parent_resource.guard_interpreter == :default
+                          Chef::GuardInterpreter::DefaultGuardInterpreter.new(command, @command_opts)
+                        else
+                          Chef::GuardInterpreter::ResourceGuardInterpreter.new(@parent_resource, command, @command_opts)
+                        end
+          interpreter.evaluate
+        end
```

Comment:
>  Right now they have implementation details specific to the fact that they are being ran as Guards.

Taking a look at those classes, it isn't clear to me what implementation details you are referring to.  From my reading, DefaultGuardInterpreter simply shells out and returns the result of `success?` on the command's status.  

ResourceGuardInterpreter is a bit more complex but it seems like the general goal of that class is already what we want here ("Run a command and tell me if it succeeded" with the caveat that in this special case we want to run the command in a context that makes sense given the parent resource).  Modulo bugs in that class, I can't really to determine what I would do to make it more general purpose.  

If we care about keeping an interface similar to not_if/only_if then I really think we should reuse this class in some form.  If we don't care about this, all of this could likely be replaced with:

```
     begin
        shell_out(command, @command_opts).status.success?
      rescue Chef::Exceptions::CommandTimeout
        Chef::Log.warn "Command '#{command}' timed out"
        false
     end
```